app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -93,6 +95,13 @@ protected void onCreate(final Bundle savedInstanceState) {
         } catch (IOException e) {
             LOGGER.info("Bluetooth handler not started, most likely Bluetooth is not enabled");
             this.enableBluetooth(false);
+        } catch (NullPointerException e) {
+            LOGGER.error("Could not start the Bluetooth service", e);
+            new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                    .setTitleText(this.getString(R.string.something_went_wrong))
+                    .setContentText(this.getString(R.string.something_went_wrong_QR_popup_explanation))
```

Comment:
This message is a bit vague, which it should be. The user should just try it again, this was because of that weird error which sometimes occurs (mostly on my mobile) which gives a nullpointer when you go back from the camera.